src/buildings/resource/building_resource.gd

```diff
@@ -3,43 +3,16 @@ class_name BuildingResource
 
 @export_group("Display")
 @export var name: String
-@export var type: Building.TYPES
+@export var type: EnumAutoload.BuildingType
 @export var sprite: Texture2D
-@export_group("Economy")
+@export_group("Production")
+@export var resource_prod: ResourceData
 @export var housing_prod: int
-@export var food_prod: int
-@export var water_prod: int
-@export var air_prod: int
-@export var metal_prod: int
+@export var storage_prod: int
 @export_group("Construction")
-@export_subgroup("Building")
-@export var food_cost: int
-@export var air_cost: int
-@export var water_cost: int
-@export var metal_cost: int
-@export var people_cost: int # required for building. Will be refund after finished building
-@export_subgroup("Destroying")
-@export var refund_food: int
-@export var refund_air: int
-@export var refund_water: int
-@export var refund_metal: int
+@export var resource_cost: ResourceData
+@export var people_cost: int # required for building. Will be refunded after finished building
 @export var refund_population: int # Exclusive to CRYO POD type building
-@export_subgroup("Timings")
 @export var construction_time: int
 @export var destruction_time: int
 
-
-func _init(
```

Comment:
Why are we removing the `_init` method here? This stops us from creating `BuildingResource` objects via code and may have some unintended interaction with resources created via the inspector.

Reply:
It will be fine don't worry, we can still create resource from code (if we ever do that), by just create it normally
`var br = BuildingResource.new()`
and then assign the variable one by one.
`br.name = "Vertical Farm`
 The _init() just let us create a new BuildingResource with arguments directly
`var br = BuildingResource.new("VerticalFarm")`
So it just slightly less code. And it won't change anything with resource created via the inspector.

I deleted it since each time we modify this, we have to update the `_init()` code even though we don't use it, which cost some time.

Reply:
:+1: